crypto/armcap.c

```diff
@@ -102,28 +102,34 @@ static unsigned long (*getauxval) (unsigned long) = NULL;
 
 void OPENSSL_cpuid_setup(void)
 {
+# ifndef OPENSSL_ARMCAP_OVERRIDE
     const char *e;
     struct sigaction ill_oact, ill_act;
     sigset_t oset;
+    volatile unsigned int mask;
+# endif
     static int trigger = 0;
 
     if (trigger)
         return;
     trigger = 1;
 
-    if ((e = getenv("OPENSSL_armcap"))) {
-        OPENSSL_armcap_P = (unsigned int)strtoul(e, NULL, 0);
-        return;
-    }
+# ifdef OPENSSL_ARMCAP_OVERRIDE
+    OPENSSL_armcap_P = OPENSSL_ARMCAP_OVERRIDE;
```

Comment:
I'm still opposed to the additional knob. Question is how many would actually use it, especially in suggested form. By this last remark I mean that it would make sense if it would also resolve run-time switches at compile time, and even remove unused code. But otherwise it would really be something for somebody to occasionally test something in non-production environment. But thing is that in such case that somebody can just as well simply temporarily modify the code by adding an assignment followed by return. [I did it at several occasions.] As for making it real in sense that capability override flag would resolve run-time switches and drop unused code. Trade-off here is if savings are worth the trouble of maintaining the #if spaghetti. I mean OpenSSL is not exactly suitable for resource-constrained environments and space savings would be negligible **in proportion** to the rest.

Reply:
For completeness sake I probably should mention that there were suggestions to take run-time switches out of assembly modules and break them into segments so that linker can omit unreferenced code. The question was risen rather in context of code reuse in other projects [more suitable for resource-constrained environments]. I'd say this would be more meaningful goal than to introduce #if spaghetti that would be OpenSSL-specific. However, such goal would have to be supported by multiple projects, there ought to be some synergy...